src/asm/main.c

```diff
@@ -84,7 +85,7 @@ static char *make_escape(char const *str)
 }
 
 /* Short options */
-static const char *optstring = "b:D:Eg:hi:LM:o:p:r:VvW:w";
+static const char *optstring = "b:D:Ef:g:hi:LM:o:p:r:VvW:w";
```

Comment:
Fixed-point precision doesn't make sense as an option—this should be removed.

Reply:
Why would it not make sense? If your program uses a single precision value (and yes, this is a real use case), the command line is a perfectly reasonable place to specify it.

Reply:
It can be specified simply as an `OPT` at the beginning of the program, and in your `defines.asm` or what have you if you really want it everywhere.

Reply:
The same argument applies to any option that `OPT` can define, though.
There's a good reason why command-line options are still allowed, and it's not just backwards compatibility.

Reply:
I would find it surprising for `f` (or `q` when that's renamed) to be the only `opt` not also available on the command line. And yeah, as a global program setting, it makes as much sense to set here as `-b`, `-g`, `-D`, `-r`, etc.

(Speaking of which, `opt r` could be useful for locally raising the recursion depth limit for particular loops, without making it unnecessarily high for the whole program.)

Reply:
I personally don't think what can be set via `OPT` should also be settable from CLI.

Reply:
But it always has before. I expect `-Q` to be as useful as `-h` (an option you're likely to set once for the whole project), and more useful than `-b` (custom binary digits are nice for the occasional inline 1bpp graphic, but not necessary for every binary literal). I don't understand what doesn't make sense about allowing `rgbasm -Q`.

Reply:
I do personally agree with ISSO; it makes it much harder to re-use code if you have to keep in mind whatever the previous project's CLI flags were. If the goal is to minimize the amount of command-line options that change the way the code is compiled in unexpected ways, it's always a good idea to start somewhere. If not, then consistency is the bigger priority.

Reply:
> I do personally agree with ISSO; it makes it much harder to re-use code if you have to keep in mind whatever the previous project's CLI flags were. If the goal is to minimize the amount of command-line options that change the way the code is compiled in unexpected ways, it's always a good idea to start somewhere. If not, then consistency is the bigger priority.

This is yet another case of "you're not allowed to write code I don't like".
If you don't want to configure these things using command-line options, _don't do that_. Tools that aim to be generally useful must not impose style constraints on projects without need.

Reply:
I want to be able to re-use code from other projects without having to study its specific build system, and likewise be able to integrate code *into* other projects without having to study its specific build system. That's the root of my concern here.

I believe that toolchains should absolutely be opinionated.

Reply:
> I believe that toolchains should absolutely be opinionated.

In that case there's no point debating. We will never agree on anything.

Reply:
I think that remark was uncalled for. I am of the firm opinion that where holding an opinion can have a measurable benefit to the community of developers as a whole, as opposed to the individual developer, that opinion is worth at the very least considering.

Reply:
It's not an insult, but a fact. We have completely different beliefs about how a toolchain should work, and thus there's little point trying to debate on what a tool should do based on completely opposed world views.

Reply:
Yes, but in the end neither of us are going to make the final decision, and with opposing viewpoints we can provide a lot of valuable insight for the people who are.

Reply:
I don't see any reason to disable it. Sure, it's not great for copy/pasting code, but that's easy to fix by saying "try not to do this please". Principle of least astonishment.

Reply:
The thing is, code already depends on CLI flags: "`mov a, %XoXoXo`" only makes sense after you see `rgbasm -D mov=ld -b oX`. And the documentation for `opt` expects this: it just says "`OPT` can be used to change some of the options during assembling from within the source, instead of defining them on the command-line", and relies on rgbasm(1) to document what each option does. If we ever want to add source-only configuration, I'd suggest a new `pragma` statement (and I don't think that's actually worth doing, `opt` works as well here as it does for `-D` or `-g`).